### PR TITLE
MPI_T MCA disabling mpiio

### PR DESCRIPTION
**Pull request by ykempf**
Here's the prototype, as this has highest precedence, do we want to check whether other ways of setting this parameter are active? Or are we happy with this printf warning/reminder? This also seems to throw an MPI error at the end of execution, at least on my laptop, needs to be investigated. Though it is whenthe job is exiting after successful completion of the simulation, so not a huge problem it seems.

```
--------------------------------------------------------------------------
A system call failed during shared memory initialization that should
not have.  It is likely that your MPI job will now either abort or
experience performance degradation.

  Local host:  lx6-flafo-03
  System call: unlink(2) /dev/shm/vader_segment.lx6-flafo-03.22f0001.1
  Error:       No such file or directory (errno 2)
--------------------------------------------------------------------------
[lx6-flafo-03:20092] *** Process received signal ***
[lx6-flafo-03:20092] Signal: Segmentation fault (11)
[lx6-flafo-03:20092] Signal code: Address not mapped (1)
[lx6-flafo-03:20092] Failing at address: 0x30
--------------------------------------------------------------------------
A system call failed during shared memory initialization that should
not have.  It is likely that your MPI job will now either abort or
experience performance degradation.

  Local host:  lx6-flafo-03
  System call: unlink(2) /dev/shm/vader_segment.lx6-flafo-03.22f0001.0
  Error:       No such file or directory (errno 2)
--------------------------------------------------------------------------
[lx6-flafo-03:20091] *** Process received signal ***
[lx6-flafo-03:20091] Signal: Segmentation fault (11)
[lx6-flafo-03:20091] Signal code: Address not mapped (1)
[lx6-flafo-03:20091] Failing at address: 0x540
--------------------------------------------------------------------------
Primary job  terminated normally, but 1 process returned
a non-zero exit code. Per user-direction, the job has been aborted.
--------------------------------------------------------------------------
--------------------------------------------------------------------------
mpirun noticed that process rank 1 with PID 0 on node lx6-flafo-03 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
```

### CODE REVIEW

**Comment by ykempf** [comment]
Related to/workaround for #698 so the users don't have to scratch their head and then set the environment or mpirun MCA flag.

**Comment by markusbattarbee** [comment]
so should we first check what type of MPI is being used? IIRC this was only required for OpenMPI, not MPICH or whatever the third somewhat common implementation was.

Strange that error message, but might be a cubbli-only thing?

**Comment by ykempf** [comment]
Probably a cubbli issue my error message, nothing suspicious on Mahti.

File: vlasiator.cpp
Comment (markusbattarbee) [inline vlasiator.cpp:312]: I'd put this inside the `if`-clause on line 272, right at the end after the `#endif`

File: vlasiator.cpp
Comment (markusbattarbee) [inline vlasiator.cpp:279]: I'd condense this a bit by removing empty lines

**Review by markusbattarbee — APPROVED** [review]

File: vlasiator.cpp
Comment (ykempf) [reply vlasiator.cpp:312]: Or can I move that before `MPI_Init_thread()`??

Comment (markusbattarbee) [reply vlasiator.cpp:312]: Ah right, I was focusing just on where the mpiioMessage is generated. I guess it's fine as is then, not worth adding a check for the length of the mpiioMessage or a comment about mpiioMessages as it's easy enough to see what it's associated with.

Comment (ykempf) [reply vlasiator.cpp:312]: I just saw that we have two calls to MPI_Comm_rank (well the second one is in the unlikely case `if (required > provided){` – can't remember that ever happening), I simplified that in the upcoming commit.

Comment (ykempf) [reply vlasiator.cpp]: Now it's simpler.

**Comment by ykempf** [comment]
So where is the report of this supposedly successful testpackage? I'd want to see it before hitting this merge button...

**Comment by ykempf** [comment]
That says "Invalid request. Only 65535 characters are allowed; 77087 were supplied."

**Comment by ykempf** [comment]
OK so testpackage report fails with that 64k limit. So here's the output file salvaged from Carrington.
[testpackage_run_output.txt](https://github.com/fmihpc/vlasiator/files/12718605/testpackage_run_output.txt)

**Comment by ykempf** [comment]
And so thereby I declare the testpackage passed and shall merge.

**Pull request merged**
